actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,33 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
```

Comment:
That'd be a nice generalization, but I think it should be accomplished in a separate PR. As it is, this is specific to the array of Symbols used to store variants.